src/main/java/com/project/Investment/App/controller/jdbc/RestControllerEntityJdbc.java

```diff
@@ -0,0 +1,65 @@
+package com.project.Investment.App.controller.jdbc;
+
+import com.project.Investment.App.dto.EntityDtoRequest;
+import com.project.Investment.App.model.Entity;
+import com.project.Investment.App.service.EntityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/entity")
+public class RestControllerEntityJdbc {
+
+    private final EntityService service;
+
+    @Autowired
+    public RestControllerEntityJdbc(EntityService service) {
+        this.service = service;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<List<Entity>> getById(
+            @PathVariable(name = "id") String id,
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) @RequestParam(value = "effectiveDate", required = false) Optional<LocalDate> effectiveDate,
+            @RequestParam(value = "limit") Optional<Integer> limit
+    ) {
+        return  ResponseEntity.status(HttpStatus.OK).body(service.findById(id, effectiveDate, limit)));
+    }
+    @GetMapping("/benchmarkId/{id}")
+    public ResponseEntity<List<Entity>> getByDefaultBenchmarkId(@PathVariable(name = "id") String id) {
+        return  ResponseEntity.status(HttpStatus.OK).body(service.findByDefaultBenchmarkId(id));
+    }
+
+    @GetMapping()
+    public ResponseEntity<List<Entity>> getAll(@RequestParam(value = "name", required = false) Optional<String> name) {
```

Comment:
You should either use `name` in filtering or remove it from this method

Reply:
Done